extensions/spatial/src/org/sbml/jsbml/xml/parsers/SpatialParser.java

```diff
@@ -241,9 +244,15 @@ public boolean processAttribute(String elementName, String attributeName,
       }
       contextObject = spatialReaction;
     }
-
-    return super.processAttribute(elementName, attributeName, value, uri, prefix,
+    
+    isAttributeRead = super.processAttribute(elementName, attributeName, value, uri, prefix,
       isLastAttribute, contextObject);
+    
+    if(!isAttributeRead) {
```

Comment:
This should not need to be done, it is already done in https://github.com/sbmlteam/jsbml/blob/8ea32d6aef94624f0b9110e8a41a58303b2a8c99/core/src/org/sbml/jsbml/xml/stax/SBMLReader.java#L1196
So in those cases, the method processUnknownAttribute would be called twice!

Reply:
@niko-rodrigue  But somehow, this was not being done automatically. Only when I added this code, it registered the unknown attribute.

Reply:
Now that your test is working, could you try to comment the call to super.processUnknownAttribute and check if the test is still working or not.